langchain_bot.py

```diff
@@ -0,0 +1,66 @@
+from langchain.document_loaders import DirectoryLoader
+
+loader = DirectoryLoader(
+    './Langchain/data/', # my local directory
+    glob='**/*.pdf',     # we only get pdfs
+    show_progress=True
+)
+docs = loader.load()
+
+
+from langchain.text_splitter import CharacterTextSplitter
+
+text_splitter = CharacterTextSplitter(
+    chunk_size=1000, 
+    chunk_overlap=0
+)
+docs_split = text_splitter.split_documents(docs)
+
+import os
+
+PINECONE_API_KEY = ... # find at app.pinecone.io
+PINECONE_ENV = ...     # next to api key in console
+OPENAI_API_KEY = ...   # found at platform.openai.com/account/api-keys
+
+os.environ['OPENAI_API_KEY'] = OPENAI_API_KEY
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Sensitive data like API keys should not be hardcoded into the source code for security reasons. Instead, consider using environment variables or a secure method to store these keys. Also, it's better to check if these keys are set before proceeding with the rest of the code.
```diff
- PINECONE_API_KEY = ... # find at app.pinecone.io
- PINECONE_ENV = ...     # next to api key in console
- OPENAI_API_KEY = ...   # found at platform.openai.com/account/api-keys
+ PINECONE_API_KEY = os.getenv('PINECONE_API_KEY')
+ PINECONE_ENV = os.getenv('PINECONE_ENV')
+ OPENAI_API_KEY = os.getenv('OPENAI_API_KEY')

+ if not all([PINECONE_API_KEY, PINECONE_ENV, OPENAI_API_KEY]):
+     raise ValueError("API keys or environment variables are not set.")
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->